src/Microsoft.Azure.WebJobs.Host.Storage/BlobStorageConcurrencyStatusRepository.cs

```diff
@@ -0,0 +1,110 @@
+
+        public BlobStorageConcurrencyStatusRepository(IConfiguration configuration, IHostIdProvider hostIdProvider, ILoggerFactory loggerFactory)
+        {
+            _configuration = configuration;
+            _hostIdProvider = hostIdProvider;
+            _logger = loggerFactory.CreateLogger(LogCategories.Scale);
+        }
+
+        internal CloudBlobContainer Container
+        {
+            get
+            {
+                if (_blobContainer == null)
+                {
+                    string storageConnectionString = _configuration.GetWebJobsConnectionString(ConnectionStringNames.Storage);
+                    _blobContainer = CreateBlobContainer(storageConnectionString);
+                }
+                return _blobContainer;
+            }
+        }
+
+        public async Task<ConcurrencyStatusSnapshot> ReadAsync()
+        {
+            string blobPath = await GetBlobPathAsync();
+            try
+            {
+                CloudBlockBlob blob = Container.GetBlockBlobReference(blobPath);
+                if (await blob.ExistsAsync())
+                {
+                    string content = await blob.DownloadTextAsync();
+                    if (!string.IsNullOrEmpty(content))
+                    {
+                        var result = JsonConvert.DeserializeObject<ConcurrencyStatusSnapshot>(content);
+                        return result;
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, $"Error reading snapshot blob {blobPath}");
+                throw e;
+            }
+
+            return null;
+        }
+
+        public async Task WriteAsync(ConcurrencyStatusSnapshot snapshot)
+        {
+            var content = JsonConvert.SerializeObject(snapshot);
+
+            string blobPath = await GetBlobPathAsync();
+            try
+            {
+                CloudBlockBlob blob = Container.GetBlockBlobReference(blobPath);
+
+                using (StreamWriter writer = new StreamWriter(await blob.OpenWriteAsync()))
+                {
+                    await writer.WriteAsync(content);
+                }
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, $"Error writing snapshot blob {blobPath}");
+                throw e;
+            }
+        }
+
+        private CloudBlobContainer CreateBlobContainer(string connectionString)
+        {
+            var account = CloudStorageAccount.Parse(connectionString);
+            var client = account.CreateCloudBlobClient();
+            var container = client.GetContainerReference(HostContainerName);
+
+            container.CreateIfNotExists();
+
+            return container;
+        }
+
+        internal async Task<string> GetBlobPathAsync()
+        {
+            string hostId = await _hostIdProvider.GetHostIdAsync(CancellationToken.None);
+            return $"scale/{hostId}/concurrencyStatus.json";
```

Comment:
Same nit about conflating scale and concurrency. It's not a blocker, but feels a little weird to me and might confuse customers who peek behind the curtains.

Reply:
Changing to use "concurrency" rather than "scale" for the path